packages/oas-resolver/index.js

```diff
@@ -139,29 +140,53 @@ function testProtocol(input, backup) {
     return 'file:';
 }
 
-function resolveExternal(root, pointer, options, callback) {
-    var u = url.parse(options.source);
-    var base = options.source.split('\\').join('/').split('/');
+function getAbsoluteRefPath(source, $ref) {
+    let base = source.split('\\').join('/').split('/');
     let doc = base.pop(); // drop the actual filename
     if (!doc) base.pop(); // in case it ended with a /
+    base = base.join('/');
+    let u = url.parse(source);
+    let u2 = url.parse($ref);
+    let effectiveProtocol = testProtocol(u2.protocol, u.protocol)
+    if (effectiveProtocol === 'file:') {
+        return {
+            target: path.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+    else {
+        return {
+            target: url.resolve(base ? base + '/' : '', $ref),
+            effectiveProtocol,
+            base
+        };
+    }
+}
+
+function resolveToName(ref, refs) {
+    let currentListOfResolved = [];
+    for (let key in refs) {
+        if (refs[key].resolvedAt) currentListOfResolved.push(refs[key].resolvedAt);
```

Comment:
I'm pretty opposed to single-line if statements. They can be harder to notice and without brackets are more prone to accidental errors when changes come along in the future.